src/Namshi/JOSE/JWT.php

```diff
@@ -34,9 +34,9 @@ class JWT
      */
     public function __construct(array $payload, array $header)
     {
-        $this->payload = $payload;
-        $this->header  = $header;
-        $this->encoder = new Base64UrlSafeEncoder();
+        $this->setPayload($payload);
```

Comment:
@brianjmiller if there is no special logic in setters and/or no need to expose them externally I would remove them.